### PR TITLE
Adding tracks environments

### PR DESCRIPTION
**Pull request by Bac0nEater**
Changed:
- added 9 tracks in meshes folder
- added 2 track environments (track 1 and track 2)
- added a parent class for all the track environments
- rename the original provided track environment to track_original

TODO:
- I will create a parent class for CarGoalEnvironment, CarWallEnvironment and CarTrackParentEnvironment because there are too many duplication.

### CODE REVIEW

**Comment by retinfai** [comment]
Parent Environment sounds good. 

I'm also wondering whether we should bundle the deletion of all `environment_training.py` files in favour of the `train.py` and `test.py` in this PR? Or should we use another PR to do this via. #49

**Comment by Bac0nEater** [comment]
If it is just deletion and no changes are needed, then we can do it in this PR.
If it is a massive change, then I reckon in another PR

**Comment by retinfai** [comment]
Well kinda. So after the "clean up" in the `reinforcement_learning` package there should only be two scripts `train.py` and `test.py` and there should only be two launch files `train.launch.py` and `test.launch.py`.  Using these 4 things, we should be able run any environment via `train.yaml` and `test.yaml` in the config directory.

It may be too many changes?

**Comment by Bac0nEater** [comment]
I see what you mean. Yeh, I reckon in another PR, if you don't mind

**Comment by retinfai** [comment]
No worries, I can work on it after this is merged in

**Comment by Bac0nEater** [comment]
Thank you! I am currently working on parent class for the environment, hopefully it won't have merge conflict

File: src/environment_interfaces/srv/Reset.srv
[the remark this answers is not in the document]
Comment (retinfai) [reply src/environment_interfaces/srv/Reset.srv:1]: I'd recommend either creating a new interface, or updating the CarGoal, CarWall, and CarBlock resets to handle the renamed request params

**Review by retinfai — CHANGES_REQUESTED** [review]

File: src/environment_interfaces/srv/Reset.srv
[the remark this answers is not in the document]
Comment (Bac0nEater) [reply src/environment_interfaces/srv/Reset.srv:1]: It won't break, that's why I left the x and y there.
I am planning to make them consistent when creating parent class for the environments.

[the remark this answers is not in the document]
Comment (retinfai) [reply src/environment_interfaces/srv/Reset.srv:1]: ohh right, I thought it got replaced. No worries here then

**Review by retinfai — APPROVED** [review]

**Pull request merged**